source/particle/generator/interface.cc

```diff
@@ -108,7 +108,7 @@ namespace aspect
       {
         // Uniform distribution on the interval [0,1]. This
         // will be used to generate random particle locations.
-        boost::uniform_01<double> uniform_distribution_01;
+        std::uniform_real_distribution uniform_distribution_01(0.0, 1.0);
```

Comment:
Allowed in C++17 but not before. You need this:
```suggestion
        std::uniform_real_distribution<double> uniform_distribution_01(0.0, 1.0);
```